framework/include/fvkernels/FVScalarLagrangeMultiplierConstraint.h

```diff
@@ -0,0 +1,38 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FVElementalKernel.h"
+
+/**
+ * Base class for implementing constraints on finite volume variable elemental values using scalar
+ * Lagrange multipliers
+ */
+class FVScalarLagrangeMultiplierConstraint : public FVElementalKernel
+{
+public:
+  static InputParameters validParams();
+
+  FVScalarLagrangeMultiplierConstraint(const InputParameters & parameters);
+
+  const MooseVariableScalar & lambdaVariable() const { return _lambda_var; }
+
+private:
+  void computeResidual() override final;
+  void computeJacobian() override final;
+  void computeOffDiagJacobian() override final;
+  ADReal computeQpResidual() override;
```

Comment:
is this used anymore? I suppose the code coverage report will tell me

Reply:
I prefer the compile time error that comes with a non-implemented pure virtual (from `FVElementalKernel`)